x-pack/legacy/plugins/file_upload/public/util/pattern_reader.js

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+const oboe = require('oboe');
+
+
+export class PatternReader {
+
+  constructor() {
+    this.oboeStream = oboe();
+  }
+
+  onGeoJSONFeaturePatternDetect = featureDetectCallback => {
+    this.oboeStream.node({
+      'features.*': feature => featureDetectCallback(feature),
+      // Handle single feature files
+      '!.geometry': (geom, path, ancestors) => {
+        const feature = ancestors[0];
+        const { geometry } = featureDetectCallback(feature);
+        return geometry;
+      }
+    });
+  }
+
+  writeDataToPatternStream = data => this.oboeStream.emit('data', data);
```

Comment:
this is a method on the type, and makes more sense on the prototype, rather than as an instance-variable.

```
writeDataPatternStream() {
...
}
```